```diff
diff --git a/pkgs/development/python-modules/pytest-markdown-docs/default.nix b/pkgs/development/python-modules/pytest-markdown-docs/default.nix
index 8b81957e78e1..e259c154f5e2 100644
--- a/pkgs/development/python-modules/pytest-markdown-docs/default.nix
+++ b/pkgs/development/python-modules/pytest-markdown-docs/default.nix
@@ -12,6 +12,7 @@
   # tests
   pytest,
   pytestCheckHook,
+  mdit-py-plugins,
 }:
 
 buildPythonPackage (finalAttrs: {
@@ -38,11 +39,9 @@ buildPythonPackage (finalAttrs: {
 
   pythonImportsCheck = [ "pytest_markdown_docs" ];
 
-  nativeCheckInputs = [ pytestCheckHook ];
-
-  disabledTests = [
-    # AssertionError: assert {'errors': 1,...pped': 0, ...} == {'errors': 0,...pped': 0, ...}
-    "test_admonition_markdown_text_file"
+  nativeCheckInputs = [
+    pytestCheckHook
+    mdit-py-plugins
   ];
 
   meta = {
```

--- a/pkgs/development/python-modules/pytest-markdown-docs/default.nix
+++ b/pkgs/development/python-modules/pytest-markdown-docs/default.nix
@@ -2,26 +2,36 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  poetry-core,
+
+  # build-system
+  hatchling,
+
+  # dependencies
   markdown-it-py,
   pytest,
+
+  # tests
+  mdit-py-plugins,
   pytestCheckHook,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "pytest-markdown-docs";
-  version = "0.5.1";
+  version = "0.9.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "modal-com";
     repo = "pytest-markdown-docs";
-    tag = "v${version}";
-    hash = "sha256-mclN28tfPcoFxswECjbrkeOI51XXSqUXfbvuSHrd7Sw=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-7fGuKTHeaMEbsHD9Zje0ODP2FRWSi0WrCZsPwRYP6rg=";
   };
 
-  build-system = [ poetry-core ];
+  build-system = [ hatchling ];
 
+  pythonRelaxDeps = [
+    "markdown-it-py"
+  ];
   dependencies = [
     markdown-it-py
     pytest
@@ -29,7 +39,10 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "pytest_markdown_docs" ];
 
-  nativeCheckInputs = [ pytestCheckHook ];
+  nativeCheckInputs = [
+    mdit-py-plugins
+    pytestCheckHook
+  ];
 
   meta = {
     description = "Run pytest on markdown code fence blocks";
@@ -37,4 +50,4 @@ buildPythonPackage rec {
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ GaetanLepage ];
   };
-}
+})

